### ISSUE TITLE
CSS Variable Based Theming

### ISSUE THREAD

**Issue by tropicaaal** [briano1905/typings#108]
CSS Custom Properties allow you to easily define values in a variable. Currently it seems that theme stylesheets just hardcode in the colors to given elements, which probably won't scale well if you plan to add new features to the site that need theming.

[Custom properties are supported on all major browsers.](https://developer.mozilla.org/en-US/docs/Web/CSS/--*#browser_compatibility)

**Comment by jakergrossman** [comment]
I agree that using CSS custom properties are great for sites with multiple themes (I've used it myself in [my portfolio page](https://github.com/jakergrossman/jakergrossman.github.io/tree/0820780faf38bbaebb9ca32728c91ae454a9b1a7/css)), though some themes may do more than just fill in the elements with colors, like [`aurora.css`](https://github.com/briano1905/typings/blob/1a4d4cb8f74870a7b185ffbcc318d5992d9d3783/themes/aurora.css#L1-L17), which uses keyframe animations to move the background gradient.

While I'm sure there is a solution that can handle things like this I'm not sure what it would look like.

Also, IIRC @briano1905 isn't very keen on adding new features to the site, but that may have changed.

**Comment by tropicaaal** [comment]
Theme's don't have to *only* be variable changes, they're still just stylesheets. Using variables will just make modifying things like colors or radii easier and less boilerplate-y.
